app/models/spy.rb

```diff
@@ -0,0 +1,55 @@
+class Spy < ActiveRecord::Base
+  has_and_belongs_to_many :marks
+  has_many :media, through: :marks
+
+  validates :uid, :username, :provider, presence: true
+  validates :uid, :username, uniqueness: true
+
+  def self.find_or_create_from_omniauth(auth_hash)
+    spy = self.find_by(uid: auth_hash["uid"], provider: auth_hash["provider"] )
+    if !spy.nil?
+      # Spy found, continue on with your life
+      return spy
+    elsif auth_hash["provider"] == "developer"
```

Comment:
These three blocks look identical to me, so could you dry it up by not doing ```elsif``` blocks, just having one assignment block instead?